src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -180,10 +181,26 @@ const convertFrameworkSummary = (framework) => {
 const convertTaskDetail = async (taskStatus, ports, logPathPrefix) => {
   // get container ports
   const containerPorts = {};
-  if (ports) {
+  const hashFunc = (str) => {
+    const hexStr = crypto.createHash('md5').update(str).digest('hex');
+    return parseInt(hexStr.substring(0, 12), 16) +
+      parseInt(hexStr.substring(12, 24), 16) + parseInt(hexStr.substring(24, 32), 16);
+  };
+  if (ports && taskStatus.attemptStatus.podUID) {
     const randomPorts = JSON.parse(ports);
-    for (let port of Object.keys(randomPorts)) {
-      containerPorts[port] = randomPorts[port].start + taskStatus.index * randomPorts[port].count;
+    if (randomPorts.ports) {
+      for (let port of Object.keys(randomPorts.ports)) {
+        const portNums = [...Array(randomPorts.ports[port].count).keys()].map((index) => {
+          const rawString = `[${taskStatus.attemptStatus.podUID}][${port}][${index}]`;
+          return hashFunc(rawString) % (randomPorts.portEnd - randomPorts.portStart) + randomPorts.portStart;
```

Comment:
>portEnd [](start = 52, length = 7)

Can we also rename portEnd and portStart in code? it is also too confusing

Reply:
Use it to pass portStart and portEnd to runtime. The point is rest-server decide the port range and pass these information to runtime. So we don't need to maintain same config in two places.
I could change the name